src/app/3d/qgs3dmapcanvaswidget.cpp

```diff
@@ -377,12 +403,55 @@ void Qgs3DMapCanvasWidget::measureLine()
   mCanvas->setMapTool( action->isChecked() ? mMapToolMeasureLine : nullptr );
 }
 
+void Qgs3DMapCanvasWidget::changePointCloudAttribute()
+{
+  QAction *action = qobject_cast<QAction *>( sender() );
+  if ( !action )
+    return;
+
+  mCanvas->setMapTool( action->isChecked() ? mMapToolPointCloudChangeAttribute : nullptr );
+}
+
 void Qgs3DMapCanvasWidget::setCanvasName( const QString &name )
 {
   mCanvasName = name;
   mDockableWidgetHelper->setWindowTitle( name );
 }
 
+void Qgs3DMapCanvasWidget::enableEditingTools( bool enable )
+{
+  mEditingToolBar->setEnabled( enable );
+}
+
+void Qgs3DMapCanvasWidget::updateLayerRelatedActions( QgsMapLayer *layer )
+{
+  if ( !layer || layer->type() != Qgis::LayerType::PointCloud )
+  {
+    enableEditingTools( false );
+
+    if ( mCanvas->mapTool() == mMapToolPointCloudChangeAttribute )
+      mCanvas->setMapTool( nullptr );
+
+    return;
+  }
+
+  QgsPointCloudLayer *pcLayer = qobject_cast<QgsPointCloudLayer *>( layer );
+  const QVector<QgsPointCloudAttribute> attributes = pcLayer->attributes().attributes();
+  const QString previousAttribute = mCboChangeAttribute->currentText();
+  whileBlocking( mCboChangeAttribute )->clear();
+  for ( const QgsPointCloudAttribute &attribute : attributes )
+  {
+    if ( attribute.name() == QLatin1String( "X" ) || attribute.name() == QLatin1String( "Y" ) || attribute.name() == QLatin1String( "Z" ) )
```

Comment:
XYZ are always uppercase?

Reply:
yes, as those values are derived from the laz file itself.